src/app/fdctl/run/tiles/fd_cswtch.c

```diff
@@ -0,0 +1,197 @@
+#include "../../../../disco/tiles.h"
+
+#include "generated/cswtch_seccomp.h"
+
+#include <unistd.h>
+#include <fcntl.h>
+#include <errno.h>
+
+#define REPORT_INTERVAL_MILLIS (100L)
+
+typedef struct {
+  long next_report_nanos;
+
+  ulong   tile_cnt;
+  int     status_fds[ FD_TILE_MAX ];
+  ulong * metrics[ FD_TILE_MAX ];
+} fd_cswtch_ctx_t;
+
+FD_FN_CONST static inline ulong
+scratch_align( void ) {
+  return 128UL;
+}
+
+FD_FN_PURE static inline ulong
+scratch_footprint( fd_topo_tile_t const * tile ) {
+  (void)tile;
+  ulong l = FD_LAYOUT_INIT;
+  l = FD_LAYOUT_APPEND( l, alignof( fd_cswtch_ctx_t ), sizeof( fd_cswtch_ctx_t ) );
+  return FD_LAYOUT_FINI( l, scratch_align() );
+}
+
+FD_FN_CONST static inline void *
+mux_ctx( void * scratch ) {
+  return (void*)fd_ulong_align_up( (ulong)scratch, alignof( fd_cswtch_ctx_t ) );
+}
+
+static void
+privileged_init( fd_topo_t *      topo,
+                 fd_topo_tile_t * tile,
+                 void *           scratch ) {
+  (void)tile;
+
+  FD_SCRATCH_ALLOC_INIT( l, scratch );
+  fd_cswtch_ctx_t * ctx = FD_SCRATCH_ALLOC_APPEND( l, alignof( fd_cswtch_ctx_t ), sizeof( fd_cswtch_ctx_t ) );
+
+  FD_TEST( topo->tile_cnt<FD_TOPO_MAX_TILES );
```

Comment:
Heads up that here the check is against `FD_TOPO_MAX_TILES` and the arrays in the context are declared with `FD_TILE_MAX`.  It's okay because `FD_TOPO_MAX_TILES < FD_TILE_MAX`, but it seems off